### PR TITLE
Bug 1527701 - Use SafeAnchor in navigation and message component

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

**Comment by punamdahiya** [comment]
@ScottDowne @Mardak PR works and successfully checks and invalidate links if not following http[s] protocol. However I am seeing an error thrown by devtools in browser console when trying to inspect respective message or navigation anchor element
Protocol error (unknownError): prop is undefined in resource://devtools/client/inspector/markup/markup.js

 Layout_endpoint :  https://gist.githubusercontent.com/punamdahiya/44f78e2f338cbf95f5c28d31faed4639/raw/1c61974ca8ebd02431512d9425031e3a9c8e04f4/test-message-feed.json

any inputs why devtools inspect is failing here? Thanks

**Comment by Mardak** [comment]
How are you selecting the anchor element? Doesn't seem to show errors in either console for me:
screen shot 2019-02-14 at 1 58 03 am

**Comment by punamdahiya** [comment]
> How are you selecting the anchor element? Doesn't seem to show errors in either console for me:
@Mardak Thanks for verifying, it seems the issue was old mozilla-central, I just tried with latest mozilla-central and that resolves the error seen earlier in devtools :)

**Comment by punamdahiya** [comment]
@Mardak Requesting review on this patch, please let me know if you don't have review bandwidths today. thanks!

File: content-src/components/DiscoveryStreamComponents/SafeAnchor/SafeAnchor.jsx [hunk 1]
```diff
@@ -19,9 +19,9 @@ export class SafeAnchor extends React.PureComponent {
   }
 
   render() {
-    const {url, className, onLinkClick} = this.props;
+    const {url, className, key, onLinkClick} = this.props;
```
Comment (Mardak) [inline content-src/components/DiscoveryStreamComponents/SafeAnchor/SafeAnchor.jsx]: Pretty sure `key` doesn't even make it to this component. We do want to set it on the component, but React handles the special prop and doesn't pass it in.

Comment (punamdahiya) [reply content-src/components/DiscoveryStreamComponents/SafeAnchor/SafeAnchor.jsx]: good to know, will remove redundant key assignment from SafeAnchor.

Comment (punamdahiya) [reply content-src/components/DiscoveryStreamComponents/SafeAnchor/SafeAnchor.jsx]: @Mardak PR updated with feedback, waiting for test to go green before merging. Thanks!

**Pull request merged**